client/app/src/services/network.service.js

```diff
@@ -347,22 +347,67 @@
 
     // Updates peer with all currency values relative to the USD price.
     function updatePeerWithCurrencies (peer, res) {
-      $http.get('https://api.fixer.io/latest?base=USD', {timeout: 2000}).then(function (result) {
-        const USD_PRICE = Number(res.data[0].price_usd)
-        var currencies = ['aud', 'brl', 'cad', 'chf', 'cny', 'eur', 'gbp', 'hkd', 'idr', 'inr', 'jpy', 'krw', 'mxn', 'rub']
-        var prices = {}
-        currencies.forEach(function (currency) {
-          prices[currency] = result.data.rates[currency.toUpperCase()] * USD_PRICE
-        })
-        prices['btc'] = res.data[0].price_btc
-        prices['usd'] = res.data[0].price_usd
-        peer.market.price = prices
-        storageService.setGlobal('peerCurrencies', prices)
+      peer = updateCurrencyConversionRates(peer)
+      const USD_PRICE = Number(res.data[0].price_usd)
+      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
+      var prices = {}
+      currencies.forEach(function (currency) {
+        prices[currency.toLowerCase()] = peer.market.conversionRates[currency] * USD_PRICE
       })
+      prices['btc'] = res.data[0].price_btc
+      prices['usd'] = res.data[0].price_usd
+      storageService.setGlobal('peerCurrencies', prices)
+      peer.market.price = prices
+    }
 
+    // Updates the currency conversion rates IF necessary
+    // Necessary if it isn't stored, or if the stored value is too old
+    function updateCurrencyConversionRates (peer) {
+      if (storageService.getGlobal('conversionRates') !== undefined) {
+        let priceObj = storageService.getGlobal('conversionRates')
+        peer.market.conversionRates = priceObj.rates
+        let storedDateString = priceObj.date
+        let storedDate = new Date(storedDateString)
+        var updateCurrencies = checkToUpdateConversionRates(storedDate)
+        if (updateCurrencies) {
+          getConversionRatesApiCall(peer)
+        }
+      } else {
+        getConversionRatesApiCall(peer)
+      }
       return peer
     }
 
+    // api call to get the conversion rates for currencies
+    function getConversionRatesApiCall (peer) {
+      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
+      var apiCall = createCurrencyConversionApiCall(currencies)
+      $http.get(apiCall, {timeout: 2000}).then(function (result) {
+        storageService.setGlobal('conversionRates', { rates: result.data.rates, date: new Date() })
+        peer.market.conversionRates = result.data.rates
+      })
+      return peer
+    }
+
+    // Checks if the stored time and the current time has crossed 4pm CET time
+    function checkToUpdateConversionRates (storedDate) {
+      const Moment = require('moment-timezone')
+      const MomentRange = require('moment-range')
```

Comment:
I think it'd make sense to move these 'require' calls out of this method. I assume it's relatively expensive to call it repeatedly.

Reply:
Agreed! made it global instead.